src/components/Axis/Axis.jsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { lucidClassNames } from '../../util/style-helpers';
+import { discreteTicks } from '../../util/chart-helpers';
+import { createClass } from '../../util/component-types';
+
+const cx = lucidClassNames.bind('&-Axis');
+
+const {
+	string,
+	array,
+	func,
+	number,
+	oneOf,
+} = React.PropTypes;
+
+/**
+ * {"categories": ["visualizations", "chart primitives"]}
+ *
+ * Axes and allies
+ *
+ * This component is a very close sister to `d3.avg.axis` and most of the logic
+ * was ported from d3.
+ */
+const Axis = createClass({
+	displayName: 'Axis',
+
+	statics: {
+		_lucidIsPrivate: true,
+	},
+
+	propTypes: {
+		/**
+		 * Appended to the component-specific class names set on the root element.
+		 */
+		className: string,
+		/**
+		 * Must be a D3 scale.
+		 */
+		scale: func.isRequired,
+		/**
+		 * Size of the ticks for each discrete tick mark.
+		 */
+		innerTickSize: number,
+		/**
+		 * Size of the tick marks found at the beginning and end of the axis. It's
+		 * common to set this to `0` to remove them.
+		 */
+		outerTickSize: number,
+		/**
+		 * An optional function that can format ticks. Generally this shouldn't be
+		 * needed since d3 has very good default formatters for most data.
+		 *
+		 * Signature: `(tick) => {}`
+		 */
+		tickFormat: func,
+		/**
+		 * If you need fine grained control over the axis ticks, you can pass them
+		 * in this array.
+		 */
+		ticks: array,
+		/**
+		 * Determines the spacing between each tick and its text.
+		 */
+		tickPadding: number,
+		/**
+		 * Determines the orientation of the ticks. `left` and `right` will
+		 * generate a vertical axis, whereas `top` and `bottom` will generate a
+		 * horizontal axis.
+		 */
+		orient: oneOf(['top', 'bottom', 'left', 'right']),
+		/**
+		 * Control the number of ticks displayed.
+		 *
+		 * If the scale is time based or linear, this number acts a "hint" per the
+		 * default behavior of D3. If it's an ordinal scale, this number is treated
+		 * as an absolute number of ticks to display and is powered by our own
+		 * utility function `discreteTicks`.
+		 */
+		tickCount: number,
+	},
+
+	getDefaultProps() {
+		return {
+			innerTickSize: 6, // same as d3
+			outerTickSize: 6, // same as d3
+			tickPadding: 3, // same as d3
+			orient: 'bottom',
+			tickCount: null,
+		};
+	},
+
+	render() {
+		const {
+			scale,
+			className,
+			orient,
+			tickCount,
+			ticks = scale.ticks
+				? scale.ticks(tickCount)
+				: discreteTicks(scale.domain(), tickCount), // ordinal scales don't have `ticks` but they do have `domains`
+			innerTickSize,
+			outerTickSize,
+			tickFormat = scale.tickFormat ? scale.tickFormat() : _.identity,
+			tickPadding,
+			...passThroughs,
+		} = this.props;
+
+		const tickSpacing = Math.max(innerTickSize, 0) + tickPadding;
+
+		// Domain
+		const range = scale.range();
+		const sign = orient === 'top' || orient === 'left' ? -1 : 1;
+		const isH = orient === 'top' || orient === 'bottom';
```

Comment:
Yep, that's right. I wanted to keep it small since I'm using in lots of small ternaries.